File: RepoTransBench/target_projects/Python/Java/antiboredom_audiogrep/src/test/java/com/antiboredom/audiogrep/public_tests/TestApiPublic.java
package com.antiboredom.audiogrep.public_tests;

import com.antiboredom.audiogrep.Audiogrep;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;
import org.mockito.Mockito;
import java.io.File;
import java.nio.file.Path;
import java.util.*;
import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

public class TestApiPublic {
    @TempDir
    Path tmpPath;

    @Test
    public void testFindFilesPublic() throws Exception {
        Audiogrep ag = new Audiogrep();
        // Use different extension and names
        File f1 = tmpPath.resolve("x.flac").toFile();
        File f2 = tmpPath.resolve("y.flac").toFile();
        File f3 = tmpPath.resolve("z.txt").toFile();
        f1.createNewFile();
        f2.createNewFile();
        f3.createNewFile();

        // For each, simulate valid
        java.nio.file.Files.write(f1.toPath(), "dummy".getBytes());
        java.nio.file.Files.write(f2.toPath(), "dummy".getBytes());
        java.nio.file.Files.write(f3.toPath(), "dummy".getBytes());

        Set<String> found = new HashSet<>(ag.findFiles(tmpPath.toString(), List.of(".flac")));
        Set<String> expected = new HashSet<>(List.of(f1.getAbsolutePath(), f2.getAbsolutePath()));
        assertEquals(expected, found);
    }

    @Test
    public void testRegexifyPublic() {
        Audiogrep ag = new Audiogrep();
        String text = "hello? world* (demo)";
        String r = ag.regexify(text);
        assertEquals("hello\\?\\ world\\*\\ \\(demo\\)", r);
    }

    @Test
    public void testGetWordTimingsPublic() throws Exception {
        Audiogrep ag = new Audiogrep();
        Path fn = tmpPath.resolve("timings_public.txt");
        String[] lines = {
            "<s> 5.0 6.0 1\n", "gamma 6.0 6.2 1\n", "zeta 6.2 6.3 1\n", "</s> 6.3 6.7 1\n"
        };
        java.nio.file.Files.write(fn, Arrays.asList(lines));
        List<List<Object>> tgt = ag.getWordTimings(fn.toString());
        assertEquals(2, tgt.size());
        assertEquals("gamma", tgt.get(0).get(0));
        assertEquals("zeta", tgt.get(1).get(0));
    }

    @Test
    public void testGroupWordsPublic() {
        Audiogrep ag = new Audiogrep();
        List<Object[]> words = List.of(
            new Object[]{"a", 1, 2, 3},
            new Object[]{"b", 2, 3, 4},
            new Object[]{"c", 3, 4, 5},
            new Object[]{"d", 4, 5, 6},
            new Object[]{"e", 5, 6, 7}
        );
        int n = 4;
        List<List<Object[]>> grouped = new ArrayList<>();
        ag.groupWords(words, n).forEachRemaining(grouped::add);
        assertEquals(2, grouped.size());
        assertEquals("a", grouped.get(0)[0][0]);
        assertEquals("b", grouped.get(1)[0][0]);
    }

    @Test
    public void testGetGroupedWordTimingsPublic() throws Exception {
        Audiogrep ag = new Audiogrep();
        Path fn = tmpPath.resolve("grouped_timings_public.txt");
        String[] lines = {
            "<s> 11.0 12.0 1\n", "x 12.0 12.44 1\n", "y 12.44 12.89 1\n", "z 12.89 13.41 1\n", "</s> 13.41 13.91 1\n"
        };
        java.nio.file.Files.write(fn, Arrays.asList(lines));
        List<List<Object[]>> groups = new ArrayList<>();
        ag.getGroupedWordTimings(fn.toString(), 2).forEachRemaining(groups::add);
        assertEquals(2, groups.size());
        assertEquals("x", groups.get(0)[0][0]);
        assertEquals("y", groups.get(1)[0][0]);
    }

    @Test
    public void testFrankenSentencePublic() {
        Audiogrep ag = new Audiogrep();
        List<List<Object[]>> inWt = List.of(
            List.of(new Object[]{"apple", 0.1, 0.2, 0}, new Object[]{"pear", 0.2, 0.3, 0}),
            List.of(new Object[]{"banana", 0.3, 0.5, 0})
        );
        List<Object> r = ag.frankenSentence("test sentence", inWt);
        assertTrue(r instanceof List);
        for (Object e : r) {
            assertTrue(e instanceof java.lang.reflect.AnnotatedElement || e instanceof Object);
        }
    }

    @Test
    public void testSearchModesPublic() throws Exception {
        Audiogrep ag = Mockito.mock(Audiogrep.class);
        Path fn = tmpPath.resolve("public.transcription.txt");
        String[] lines = {
            "<s> 3.0 3.7 1\n", "foo 3.7 3.8 1\n", "bar 3.8 4.1 1\n", "</s> 4.1 4.5 1\n"
        };
        java.nio.file.Files.write(fn, Arrays.asList(lines));
        // monkeypatch distinct outputs per mode
        when(ag.fragmentSearch(anyString(), anyList(), anyBoolean()))
            .thenReturn(List.of(Map.of("X","Y")));
        when(ag.wordSearch(anyString(), anyList(), anyBoolean()))
            .thenReturn(List.of(Map.of("Q",2)));
        when(ag.frankenSentence(anyString(), anyList()))
            .thenReturn(List.of((Object)42));
        assertEquals(List.of(Map.of("X", "Y")), ag.search("any", fn.toString(), "fragment", false));
        assertEquals(List.of(Map.of("Q", 2)), ag.search("hello", fn.toString(), "word", false));
        assertEquals(List.of(42), ag.search("repeat", fn.toString(), "sentence", false));
    }

    @Test
    public void testMakeSplicePublic() {
        Audiogrep ag = new Audiogrep();
        Path outMp3 = tmpPath.resolve("f.spliced.mp3");
        List<Object> slices = List.of(new Object(), new Object());
        try {
            ag.makeSplice("dummy.mp3", slices, outMp3.toString());
            fail("Expected exception (no ffmpeg or file error) was not thrown");
        } catch (Exception e) {
            // Should raise some exception
            assertTrue(e != null);
        }
    }
}

File: RepoTransBench/source_projects/Python/antiboredom_audiogrep/audiogrep/tests_public/test_api_public.py
import audiogrep
import pytest

def test_find_files_public(tmp_path):
    # Use different extension and names than original test
    files = [
        tmp_path / "x.flac",
        tmp_path / "y.flac",
        tmp_path / "z.txt",
    ]
    for f in files:
        f.write_text("dummy")

    found = set(audiogrep.find_files(str(tmp_path), exts=[".flac"]))
    # Should only return .flac files
    assert set(found) == {str(tmp_path / "x.flac"), str(tmp_path / "y.flac")}

def test_regexify_public():
    # Use a string not in original test
    text = "hello? world* (demo)"
    r = audiogrep.regexify(text)
    # The result should be a regex string that escapes special chars
    assert r == r"hello\?\ world\*\ \(demo\)"

def test_get_word_timings_public(tmp_path):
    # Use different lines
    fn = tmp_path / "timings_public.txt"
    lines = [
        "<s> 5.0 6.0 1\n", 
        "gamma 6.0 6.2 1\n", 
        "zeta 6.2 6.3 1\n", 
        "</s> 6.3 6.7 1\n"
    ]
    fn.write_text("".join(lines))
    tgt = audiogrep.get_word_timings(str(fn))
    assert len(tgt) == 2  # Should return only middle words
    assert tgt[0][0] == "gamma"
    assert tgt[1][0] == "zeta"

def test_group_words_public():
    # Different sequence and group size
    words = [('a', 1, 2, 3), ('b', 2, 3, 4), ('c', 3, 4, 5), ('d', 4, 5, 6), ('e', 5, 6, 7)]
    n = 4
    grouped = list(audiogrep.group_words(words, n))
    assert len(grouped) == 2  # ('a','b','c','d'), ('b','c','d','e')
    assert grouped[0][0][0] == 'a'
    assert grouped[1][0][0] == 'b'

def test_get_grouped_word_timings_public(tmp_path):
    fn = tmp_path / "grouped_timings_public.txt"
    lines = [
        "<s> 11.0 12.0 1\n", 
        "x 12.0 12.44 1\n", 
        "y 12.44 12.89 1\n", 
        "z 12.89 13.41 1\n", 
        "</s> 13.41 13.91 1\n"
    ]
    fn.write_text("".join(lines))
    groups = list(audiogrep.get_grouped_word_timings(str(fn), n=2))
    assert len(groups) == 2  # ("x","y"), ("y","z")
    assert groups[0][0][0] == "x"
    assert groups[1][0][0] == "y"

def test_franken_sentence_public():
    # Use different word timings and structure
    in_wt = [
        [("apple", 0.1, 0.2, 0), ("pear", 0.2, 0.3, 0)],
        [("banana", 0.3, 0.5, 0)]
    ]
    r = audiogrep.franken_sentence("test sentence", in_wt)
    assert isinstance(r, list)
    # Should produce a list with slice-like elements for each group
    assert all(isinstance(e, slice) or isinstance(e, tuple) for e in r)

def test_search_modes_public(monkeypatch, tmp_path):
    fn = tmp_path / "public.transcription.txt"
    lines = [
        "<s> 3.0 3.7 1\n", "foo 3.7 3.8 1\n", "bar 3.8 4.1 1\n", "</s> 4.1 4.5 1\n"
    ]
    fn.write_text("".join(lines))

    # Patch the searched methods to return distinct-dummy-for-public-tests
    monkeypatch.setattr(audiogrep, "fragment_search", lambda q,s,reg: [{"X": "Y"}])
    monkeypatch.setattr(audiogrep, "word_search", lambda q,s,reg: [{"Q": 2}])
    monkeypatch.setattr(audiogrep, "franken_sentence", lambda q,fs: [42])

    # The actual audiogrep.search should dispatch
    assert audiogrep.search("any", str(fn), "fragment", False) == [{"X": "Y"}]
    assert audiogrep.search("hello", str(fn), "word", False) == [{"Q": 2}]
    assert audiogrep.search("repeat", str(fn), "sentence", False) == [42]

def test_make_splice_public(tmp_path):
    # Create dummy slices and output file
    out_mp3 = tmp_path / "f.spliced.mp3"
    slices = [slice(0, 2), slice(3, 5)]
    # Should not actually create audio, but function should run
    with pytest.raises(Exception):  # It will likely fail (no ffmpeg)
        audiogrep.make_splice("dummy.mp3", slices, str(out_mp3))